packages/waku/src/lib/builder/build.ts

```diff
@@ -474,18 +483,24 @@ const emitHtmlFiles = async (
   distEntries: EntriesPrd,
   buildConfig: Awaited<ReturnType<typeof getBuildConfig>>,
   getClientModules: (input: string) => string[],
+  clientBuildCss: string[],
   ssr: boolean,
 ) => {
+  const css = clientBuildCss.map((l) => `<link rel="stylesheet" href="${l}">`);
+
   const basePrefix = config.basePath + config.rscPath + '/';
   const publicIndexHtmlFile = joinPath(
     rootDir,
     config.distDir,
     config.publicDir,
     config.indexHtml,
   );
-  const publicIndexHtml = await readFile(publicIndexHtmlFile, {
+  let publicIndexHtml = await readFile(publicIndexHtmlFile, {
     encoding: 'utf8',
   });
+  publicIndexHtml = publicIndexHtml.replace(/<\/head>/, `${css}</head>`);
+  await writeFile(publicIndexHtmlFile, publicIndexHtml);
+
```

Comment:
This looks really bad. Too hacky. It should already be modified in advance. Can you do things in `rscIndexPlugin`?

Reply:
Tried this, but could not, rollup should finish analyzing and creating an output. Then we'd know if there was any CSS emitted. What do you think? unless we create another build just for analyzing (which is too much as it's a complete build in itself)

Reply:
Can we do it in rscAnalyzePlugin?

Reply:
let me see

Reply:
No:
1- We haven't found the clientEntries even yet. That's the plugin's job to do so.
2- We stop walking the client files anyway.

Reply:
In the future, we need to 2-pass analyze anyway. So, let's wait this PR until then, or we do the 2-pass analyze in advance (in a different PR).

Reply:
I'll do it in another PR after #552